engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -191,32 +204,24 @@ class WorkflowManagerActor(config: Config)
       logger.info(s"$tag transitioning from $fromState to $toState")
   }
 
-  /** Submit the workflow and return an updated copy of the state data reflecting the addition of a
+  /**
+    * Submit the workflow and return an updated copy of the state data reflecting the addition of a
     * Workflow ID -> WorkflowActorRef entry.
     */
-  private def submitWorkflow(source: WorkflowSourceFiles, replyTo: Option[ActorRef], id: Option[WorkflowId]): WorkflowIdToActorRef = {
-    val workflowId: WorkflowId = id.getOrElse(WorkflowId.randomId())
-    val isRestart = id.isDefined
+  private def submitWorkflow(workflow: WorkflowStore.WorkflowToStart): WorkflowIdToActorRef = {
+    val workflowId = workflow.id
 
-    val startMode = if (isRestart) {
+    val startMode = if (workflow.state == WorkflowStore.Restartable) {
```

Comment:
Yeah I know, it just seems a little weird in code.  I'll probably get used to it.  😄